src/Adapter/Order/OrderDetailUpdater.php

```diff
@@ -80,21 +80,29 @@ public function __construct(
      * @param Order $order
      * @param Number $priceTaxExcluded
      * @param Number $priceTaxIncluded
+     * @param Number $ecotax
      *
      * @throws OrderException
      */
     public function updateOrderDetail(
         OrderDetail $orderDetail,
         Order $order,
         Number $priceTaxExcluded,
-        Number $priceTaxIncluded
+        Number $priceTaxIncluded,
+        Number $ecotax
     ): void {
         list($roundType, $computingPrecision, $taxAddress) = $this->prepareOrderContext($order);
 
         try {
+            $priceTaxExcluded = $priceTaxExcluded->minus($ecotax);
```

Comment:
This is definitely worth some extra comments to explain that (for some very weird reasons) Ecotax is added in both prices in OrderDetail (tax excluded and tax inxluded) although we don't understand why and it seems a very weird behaviour

And while we're talking about this weird behaviour I'd like some feedback about this from @PrestaShop/product-team and some clear specification (wether we keep this behaviour or not) because it makes no sense to me Mayt it's a very old bug that we are just forcibly re developing just to match the legacy behaviour... (this kind of strategy has proven to be wrong in many many cases 😅 )

Reply:
I currently add comments.

For explanation : Prices coming from the backoffice : they are displayed with ecotax so we need to remove ecotax before having precise price